Make XConn::atom_prop() work for arbitrary atoms

psychon opened the pull request:
Before this commit XcbConnection::atom_prop() used Atom::from_str(name)?
on its argument. Thus, it only worked for the predefined atoms. This
commit changes it to also work for arbitrary atoms. To make this work,
the XcbApi trait had to be changed accordingly.

Fixes: https://github.com/sminez/penrose/issues/112
Signed-off-by: Uli Schlachter <psychon@znc.in>

psychon commented:
I don't know what dangerjs is / does, but I fail to understand that check. I would guess that it just does not work for PRs...?

sminez commented on src/xcb/xconn.rs:
```diff
@@ -70,7 +70,7 @@ impl XcbConnection {
 
     fn window_has_type_in(&self, id: WinId, win_types: &[u32]) -> bool {
-        if let Ok(atom) = self.api.get_atom_prop(id, Atom::NetWmWindowType) {
+        if let Ok(atom) = self.api.get_atom_prop(id, "_NET_WM_WINDOW_TYPE") {
```
can this be changed to `Atom::NetWmWindowType.as_ref()` like the rest of the code base please: I want to avoid any hard coded `&str` literals for known atom names.

psychon replied:
Done. I amended that into the commit.

Yeah, I was not totally happy with that solution either, but obviously didn't think enough about it. Sorry.

sminez replied:
Thanks! If you can rebase off of current develop I'll get this merged :+1:

psychon replied:
Done

sminez reviewed (changes requested):
Looks like your branch needs rebasing from develop as well

sminez commented:
[TIL](https://github.com/actions/first-interaction/issues/10) that github actions downgrades the scope of the credentials given to your action when running from a fork :disappointed: 

`danger.js` runs some additional checks on the PR and then _should_ add labels to the PR based on what parts of the code base are being modified. I'll have to look into what (if anything) can be done for PRs raised from forks rather than a branch in the repo: for PR branches `danger` works just fine.

psychon committed:
Make XConn::atom_prop() work for arbitrary atoms

Before this commit XcbConnection::atom_prop() used Atom::from_str(name)?
on its argument. Thus, it only worked for the predefined atoms. This
commit changes it to also work for arbitrary atoms. To make this work,
the XcbApi trait had to be changed accordingly.

Fixes: https://github.com/sminez/penrose/issues/112
Signed-off-by: Uli Schlachter <psychon@znc.in>

sminez commented:
Oh good! The PR bot works properly now for forks as well :tada:
